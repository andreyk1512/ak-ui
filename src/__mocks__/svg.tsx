import {forwardRef} from 'react';

// eslint-disable-next-line react/display-name
const SvgrMock = forwardRef<HTMLSpanElement>((props, ref) => <span ref={ref} {...props} />);

export default SvgrMock;
