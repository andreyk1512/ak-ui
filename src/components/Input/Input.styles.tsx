import { css } from '@emotion/react';
import * as styles from '../../styles';

export const inputWrapperStyles = () => css`
  display: inline-flex;
  align-items: center;
  background-color: ${styles.color.aliceBlue};
  border-radius: 6px;
  color: ${styles.color.uclaBlue};
  width: 100%;
`;

export const inputStyles = () => css`
  display: block;
  height: 45px;
  border: none;
  box-sizing: border-box;
  padding: 0 15px;
  font-size: ${styles.fontSize.regular};
  background-color: transparent;
  overflow: hidden;
  color: inherit;
  width: 100%;

  &:focus {
    color: ${styles.color.yankeesBlue};
    outline: none;
  }
`;

export const suffixIconStyles = () => css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  margin-inline-end: 15px;
  color: inherit;
  max-width: 24px;
  max-height: 24px;
`;
