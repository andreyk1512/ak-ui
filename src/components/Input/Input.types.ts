import { InputHTMLAttributes, ReactNode } from 'react';

export type InputProps = InputHTMLAttributes<HTMLInputElement> & {
  suffixIcon?: ReactNode;
};
