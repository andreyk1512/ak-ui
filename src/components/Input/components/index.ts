export * from './Password';
export * from './TextArea';
export * from './File';
