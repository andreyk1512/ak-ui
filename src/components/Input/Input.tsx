import { Password, TextArea } from './components';
import { inputStyles, inputWrapperStyles, suffixIconStyles } from './Input.styles';
import { InputProps } from './Input.types';

export const Input = ({ suffixIcon, ...inputProps }: InputProps) => (
  <div css={inputWrapperStyles()}>
    <input {...inputProps} css={inputStyles()} />
    {suffixIcon ? <span css={suffixIconStyles()}>{suffixIcon}</span> : null}
  </div>
);

Input.Password = Password;
Input.TextArea = TextArea;
