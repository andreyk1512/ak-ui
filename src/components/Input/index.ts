export * from './Input';
export * from './Input.types';

import { Password, TextArea, File } from './components';
import { Input as OriginInput } from './Input';

export type InputProps = typeof OriginInput & {
  Password: typeof Password;
  TextArea: typeof TextArea;
  File: typeof File;
};

export const Input = OriginInput as InputProps;
Input.Password = Password;
Input.TextArea = TextArea;
Input.File = File;
