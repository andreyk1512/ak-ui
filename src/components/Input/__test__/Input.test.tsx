import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { ReactComponent as EyeSVG } from '../../../assets/icons/eye.svg';
import { Input } from '../Input';

describe('Input', () => {
  const onChange = jest.fn();

  it('should render the input element', () => {
    const textValue = 'This is a text component';
    const { container } = render(<Input value={textValue} onChange={onChange} />);

    const text = screen.getByRole('textbox');
    expect(text).toBeInTheDocument();
    expect(text).toHaveAttribute('value', textValue);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the input element with suffix icon', () => {
    const { container } = render(<Input suffixIcon={<EyeSVG title="eye" />} onChange={onChange} />);

    const text = screen.getByRole('textbox');
    const icon = screen.getByTitle('eye');
    expect(text).toBeInTheDocument();
    expect(icon).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
