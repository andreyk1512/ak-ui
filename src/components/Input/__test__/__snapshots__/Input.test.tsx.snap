// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render the input element 1`] = `
<div
  class="css-1i5hjp5"
>
  <input
    class="css-stxuvm"
    value="This is a text component"
  />
</div>
`;

exports[`Input should render the input element with suffix icon 1`] = `
<div
  class="css-1i5hjp5"
>
  <input
    class="css-stxuvm"
  />
  <span
    class="css-1a5cg1q"
  >
    <span
      title="eye"
    />
  </span>
</div>
`;
